crates/sargon/src/system/sargon_os/sargon_os_signing.rs

```diff
@@ -0,0 +1,281 @@
+use crate::prelude::*;
+use std::ops::Index;
+
+// ==================
+// Sign Signables
+// ==================
+impl SargonOS {
+    pub async fn sign_transaction(
+        &self,
+        transaction_intent: TransactionIntent,
+        role_kind: RoleKind,
+    ) -> SignedOutcome<TransactionIntent> {
+        self.sign(
+            transaction_intent.clone(),
+            self.sign_transactions_interactor(),
+            role_kind,
+        )
+        .await
+    }
+
+    pub async fn sign_subintent(
+        &self,
+        subintent: Subintent,
+        role_kind: RoleKind,
+    ) -> SignedOutcome<Subintent> {
+        self.sign(
+            subintent.clone(),
+            self.sign_subintents_interactor(),
+            role_kind,
+        )
+        .await
+    }
+
+    async fn sign<S: Signable>(
+        &self,
+        signable: S,
+        sign_interactor: Arc<dyn SignInteractor<S>>,
+        role_kind: RoleKind,
+    ) -> SignedOutcome<S> {
+        let Some(profile) = &self
+            .profile_state_holder
+            .profile()
+            .inspect_err(|error| {
+                error!("Could not initiate signing due to error: {error}")
+            })
+            .ok()
+        else {
+            return SignedOutcome::Rejected;
```

Comment:
maybe a specfic erro variant other than `Rejected` here? `MissingProfile` or so?

Reply:
Do we care to be specific? If so a result type would be more appropriate?

Reply:
(not too important)

Reply:
I think it is important we dont call internal errors for USER ERROR. `Reject` is that the user explicitly stopped signing.